pkg/csi/nodeserver.go

```diff
@@ -118,10 +118,19 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	volCap := req.GetVolumeCapability()
 	switch volumeType {
 	case LvmVolumeType:
-		err := ns.mountLvm(ctx, req)
-		if err != nil {
-			log.Errorf("NodePublishVolume: mount lvm volume %s with path %s with error: %v", req.VolumeId, targetPath, err)
-			return nil, err
+		switch volCap.GetAccessType().(type) {
+		case *csi.VolumeCapability_Block:
+			err := ns.mountLvmBlock(ctx, req)
+			if err != nil {
+				log.Errorf("NodePublishVolume(mountLvmBlock): mount lvm volume %s with path %s with error: %v", req.VolumeId, targetPath, err)
+				return nil, err
```

Comment:
please return the wrapped error, otherwise the caller may not understand the detailed error instance.

Reply:
done